lib/resque/worker.rb

```diff
@@ -491,20 +491,18 @@ def self.all_heartbeats
     # Returns a list of workers that have sent a heartbeat in the past, but which
     # already expired (does NOT include workers that have never sent a heartbeat at all).
     def self.all_workers_with_expired_heartbeats
-      workers = Worker.all
       heartbeats = Worker.all_heartbeats
       now = data_store.server_time
 
-      workers.select do |worker|
-        id = worker.to_s
-        heartbeat = heartbeats[id]
-
+      heartbeats.select {|id, heartbeat|
         if heartbeat
           seconds_since_heartbeat = (now - Time.parse(heartbeat)).to_i
           seconds_since_heartbeat > Resque.prune_interval
         else
           false
         end
+      }.each_key.map do |id|
+        find(id, :skip_exists => true)
```

Comment:
Would be good to add a one-line comment here about why we don't need to check if the worker exists.